### ISSUE TITLE
Relax RCS1198 "Avoid unnecessary boxing of value type." on string interpolation

### ISSUE THREAD

**Issue by Arthri** [dotnet/roslynator#1494]
**Product and Version Used**: Roslynator.Analyzers v4.12.3

**Steps to Reproduce**:
```cs
_ = $"{int.Parse("12")}";
```

**Actual Behavior**:
```csharp
       _______________ : RCS1198: Avoid unnecessary boxing of value type.
_ = $"{int.Parse("12")}";
```

**Expected Behavior**:
```csharp
// <no diagnostics>
_ = $"{int.Parse("12")}";
```
On C#10 and later projects, string interpolation uses a builder, which usually has a generic [AppendFormatted](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.defaultinterpolatedstringhandler.appendformatted?view=net-8.0#system-runtime-compilerservices-defaultinterpolatedstringhandler-appendformatted-1(-0)) to prevent boxing, so the analyzer is reporting a false positive.

**Comment by prezaei** [comment]
Yes, this should be avoided there. I see plenty of examples when I enable it too. 
It also seems to add it to nullable strings:

before the analyzer fix was run:

`$"{tenantId}:{objectId}"`

after it was run:

`$"{tenantId}:{objectId?.ToString()}"`

**Comment by prezaei** [comment]
Hey, I am still hitting this exact issue. An example:

<img width="540" height="100" alt="Image" src="https://github.com/user-attachments/assets/d18b98ab-901c-4b33-af00-4c1ae877a0fa" />
